src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -501,4 +558,34 @@ private function toIterator($files)
 
         return $files;
     }
+
+    /**
+     * Gets the scheme part of a filename if present or false otherwise (e.g. file:///tmp -> file).
+     *
+     * @param string $filename The filename to be parsed.
+     *
+     * @return string|bool The filename scheme if present or false.
+     */
+    private function getScheme($filename)
+    {
+        $schemeSeparator = '://';
+        $components = explode($schemeSeparator, $filename, 2);
+
+        return count($components) >= 2 ? $components[0] : false;
+    }
+
+    /**
+     * Gets the hierarchy part of a filename (e.g. file:///tmp -> /tmp).
+     *
+     * @param string $filename The filename to be parsed.
+     *
+     * @return string The filename hierarchy.
+     */
+    private function getHierarchy($filename)
```

Comment:
Includes the authority as well as the path.

STD 66 and RFC 3986 call it "hierarchical part" but hierarchy seemed nicer.

http://en.wikipedia.org/wiki/URI_scheme

Cheers